framework/src/transfers/MultiAppPostprocessorTransfer.C

```diff
@@ -70,15 +70,33 @@ MultiAppPostprocessorTransfer::execute()
     case BETWEEN_MULTIAPP:
       for (unsigned int i = 0; i < getFromMultiApp()->numGlobalApps(); i++)
       {
+        // Get source postprocessor value
+        Real pp_value = std::numeric_limits<Real>::max();
         if (getFromMultiApp()->hasLocalApp(i))
         {
-          // Get source postprocessor value
           FEProblemBase & from_problem = getFromMultiApp()->appProblemBase(i);
-          const Real & pp_value = from_problem.getPostprocessorValueByName(_from_pp_name);
-
-          if (getToMultiApp()->hasLocalApp(i))
-            getToMultiApp()->appProblemBase(i).setPostprocessorValueByName(_to_pp_name, pp_value);
+          pp_value = from_problem.getPostprocessorValueByName(_from_pp_name);
         }
+
+        // Find the postprocessor value from another process
+        if (getFromMultiApp()->numGlobalApps() == 1)
+          _communicator.min(pp_value);
+        else if (pp_value == std::numeric_limits<Real>::max() && getToMultiApp()->hasLocalApp(i))
+          mooseError("Source and target app parallel distribution must be the same");
```

Comment:
can we just do an assertion since you have that check function?

Reply:
done